Mathlib/RingTheory/MvPowerSeries/LinearTopology.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos-Fernández
+-/
+import Mathlib.Data.Finsupp.Interval
+import Mathlib.RingTheory.MvPowerSeries.PiTopology
+import Mathlib.Topology.Algebra.LinearTopology
+import Mathlib.Topology.Algebra.Nonarchimedean.Bases
+import Mathlib.RingTheory.TwoSidedIdeal.Operations
+
+/-! # Linear topology on the ring of multivariate power series
+
+- `MvPowerSeries.basis`: the ideals of the ring of multivariate power series
+all coefficients the exponent of which is smaller than some bound vanish.
+
+## Instances :
+
+- `MvPowerSeries.isLinearTopology` : if `R` has a linear topology,
+then the product topology on `MvPowerSeries σ R` is a linear topology.
+This applies in particular when `R` has a discrete topology.
+
```

Comment:
What do you think about adding to the docstring a mention of the following alternative approach?

If we had an analogue of [PolynomialModule](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/Module/Basic.html#PolynomialModule) for power series, meaning that we could consider the `R[[X]]`-module `M[[X]]` when `M` is an `R`-module, one could prove that `M[[X]]` is linearly topologized over `R[[X]]` if `M` is over `R`. To recover the ring case, it would remain to show that the isomorphism between `Rᵐᵒᵖ[[X]]` and `R[[X]]ᵐᵒᵖ` identifies their respective actions on `R[[X]]`.

Reply:
Is it really the same as the stuff above, which says that `R[[X]]` is linearly topologised over `R`?

Reply:
Your PR proves that `R[[X]]` is linearly topologized over `R[[X]]`, not only `R`, right?

Reply:
Insert the meme by Anouk Ricard